### PR TITLE
Implement an UpDownCounter Instrument for the Metrics API

### PR DESCRIPTION
**Pull request by zsistla**
* Initial UpDownCounter commit.
* Unit tests included.

`UpDownCounter` extends `AbstractMetric` from the `Metrics api` [pull request](https://github.com/open-telemetry/opentelemetry-php/pull/155/files).

### CODE REVIEW

**Comment by bobstrecansky** [comment]
@zsistla - The phan checks are failing for this PR.  You should be able to run a `make phan` and see the things you need to change locally.

You can also see the output from that phan run in the github action:
https://github.com/open-telemetry/opentelemetry-php/pull/157/checks?check_run_id=913193023

**Comment by zsistla** [comment]
Hi Bob,

Thank you for checking out the PR.

The phan errors are related to this not being a standalone PR but dependent on Artem's PR. The classes it's not detecting are from his PR.

The errors are such that I think when Artem's PR is merged, the phan checks will pass.

sdk/Metrics/UpDownCounter.php:23 PhanUndeclaredExtendedClass Class extends undeclared class \OpenTelemetry\Sdk\Metrics\AbstractMetric
sdk/Metrics/UpDownCounter.php:23 PhanUndeclaredInterface Class implements undeclared interface \OpenTelemetry\Metrics\LabelableMetric
sdk/Metrics/UpDownCounter.php:25 PhanUndeclaredTrait Class uses undeclared trait \OpenTelemetry\Sdk\Metrics\HasLabels
sdk/Metrics/UpDownCounter.php:39 PhanUndeclaredClassConstant Reference to constant UP_DOWN_COUNTER from undeclared class \OpenTelemetry\Metrics\MetricKind
[debug] Restarted process exited 1

**Comment by bobstrecansky** [comment]
I think this is ready to go after the merge conflicts are fixed!

File: sdk/Metrics/UpDownCounter.php [hunk 1]
```diff
@@ -16,6 +32,7 @@ class UpDownCounter extends Counter implements API\UpDownCounter
     /**
      * Get $type
      *
+     * @access	public
      * @return  int
```
Comment (bobstrecansky) [inline sdk/Metrics/UpDownCounter.php:36]: Looks like these two docblocks don't line up?  Looks like one is spaced and the other is tabbed?

**Pull request merged**